modules/migration-cleaning-up-failed-migration.adoc

```diff
@@ -0,0 +1,132 @@
++
+The following resources must be deleted:
+
+* rsync/stunnel pods, both in the source and target namespaces
+* Secrets and configmap, both in the source and target namespaces
+* Route and service only in the target namespaces
+
+
+
+
+. Unquiescing applications
++
+If your application was quiesced during migration, you must unquiesce it by scaling it back to its initial replica count. The original replica count is indicated in a label that MTC adds to the `deployment` resource when a source application is quiesced during migration:
+
++
+----
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+ annotations:
+    deployment.kubernetes.io/revision: "1"
+    migration.openshift.io/preQuiesceReplicas: "1"
+----
+Unquiesce the application in either of the following two ways:
+
+** Manually edit the deployment primitive (`Deployment`, `DeploymentConfig`, etc.) and set the `spec.replicas` field back to its original, non-zero value:
++
+----
+$ oc edit deployment <deployment_name>
+----
+
+** Scale your deployment with the `oc scale` command:
++
+----
+$ oc scale deployment <deployment_name> --replicas=<desired_replicas>
+----
+
+
+. Deleting the MTC Operator and cluster-scoped resources
+
++
+CAUTION: This procedure entirely removes Velero. Perform it only if there are no more Velero instances employed on the cluster.
+
+.. Delete the Migration Controller and its resources:
++
+[source,terminal]
+----
+$ oc delete migrationcontroller <resource_name>
+----
++
+Wait for the MTC Operator to finish deleting the resources.
+
+.. Uninstall the MTC Operator:
+
+   ** OpenShift 4: Uninstall the Operator in the [web console](https://docs.openshift.com/container-platform/4.7/operators/olm-deleting-operators-from-cluster.html) LINK ERROR or by running the following command:
++
+----
+$ oc delete ns openshift-migration
+----
+
+   ** OpenShift 3: Uninstall the operator by deleting it:
++
+----
+$ oc delete -f operator.yml
+----
+
+.. Delete the cluster-scoped resources:
+
+   ** Migration custom resource definition:
++
+----
+$ oc delete $(oc get crds -o name | grep 'migration.openshift.io')
+----
+   ** Velero custom resource definition:
++
+----
+$ oc delete $(oc get crds -o name | grep 'velero')
+----
+   ** Migration cluster role:
++
+----
+$ oc delete $(oc get clusterroles -o name | grep 'migration.openshift.io')
+----
+   ** Migration-operator cluster role:
++
+----
+$ oc delete clusterrole migration-operator
+----
+   ** Velero cluster role:
++
+----
+$ oc delete $(oc get clusterroles -o name | grep 'velero')
+----
+   ** Migration cluster role bindings:
++
+----
+$ oc delete $(oc get clusterrolebindings -o name | grep 'migration.openshift.io')
+----
+   ** Migration-operator cluster role bindings:
++
+----
+$ oc delete clusterrolebindings migration-operator
+----
+   ** Velero cluster role bindings:
++
+----
+$ oc delete $(oc get clusterrolebindings -o name | grep 'velero')
+----
```

Comment:
Principles of minimalism: Include a verification step so the user can tell that the cleanup was successful.